mmv1/third_party/terraform/resources/resource_container_cluster.go.erb

```diff
@@ -448,18 +455,21 @@ func resourceContainerCluster() *schema.Resource {
 				Optional:    true,
 				Computed:    true,
 				Description: `Per-cluster configuration of Node Auto-Provisioning with Cluster Autoscaler to automatically adjust the size of the cluster and create/delete node pools based on the current needs of the cluster's workload. See the guide to using Node Auto-Provisioning for more details.`,
-				ConflictsWith: []string{"enable_autopilot"},
 				Elem: &schema.Resource{
 					Schema: map[string]*schema.Schema{
 						"enabled": {
-							Type:        schema.TypeBool,
-							Required:    true,
+							Type:          schema.TypeBool,
```

Comment:
I understand that making this optional will simplify user experience, but allowing empty block `cluster_autoscaling{ }` to be specified usually causes more problem in Terraform. Therefore I would recommend that we keep this field as required.

Reply:
I'm happy to make that change, but we're seeing a lot of friction with terraform's client side validation. It keeps causing unnecessary delays in support for new features. Especially when those features are ~support for existing fields (like this one).